src/bundle/DependencyInjection/Compiler/ClassNameResourceNamePass.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\Rest\DependencyInjection\Compiler;
+
+use Ibexa\Bundle\Rest\ApiPlatform\ClassNameResourceNameCollectionFactory;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+final readonly class ClassNameResourceNamePass implements CompilerPassInterface
+{
+    public const string API_PLATFORM_RESOURCE_SERVICE_TAG = 'ibexa.api_platform.resource';
+
+    public function process(ContainerBuilder $container): void
```

Comment:
I think we should be able to get rid of this compiler pass using native DI component features, but this could be evaluated as follow up.